server/middleware/per-request.js

```diff
@@ -31,6 +31,16 @@ function perRequestContextFactory(options) {
   var scope = options.name || name;
   var enableHttpContext = options.enableHttpContext || false;
   var ns = LoopBackContext.createContext(scope);
+  // Monkey-patch LoopBackContext.getCurrentContext in order to fix issue #17
```

Comment:
I tried to bind inside the original method, but for some reasons it didn't work.
It should work if I monkey-patch the method inside, say, a utility module. This will be my next attempt, unless I figure out why it doesn't work in the original method. Please wait for the commit.
I agree that it needs to be moved away from the middleware. In fact, if you add the middleware to the app, it works (even outside the middleware); but if you don't add the middleware to the app... whoops.